tests/test_config.py

```diff
@@ -80,9 +81,156 @@ def test_config_should_reload_is_set(
     assert config_reload.should_reload is expected_should_reload
 
 
-def test_reload_dir_is_set() -> None:
-    config = Config(app=asgi_app, reload=True, reload_dirs="reload_me")
-    assert config.reload_dirs == ["reload_me"]
+def test_should_warn_on_invalid_reload_configuration(
+    tmp_path: Path, caplog: pytest.LogCaptureFixture
+) -> None:
+    config_class = Config(app=asgi_app, reload_dirs=[str(tmp_path)])
+    assert not config_class.should_reload
+    assert len(caplog.records) == 1
+    assert (
+        caplog.records[-1].message
+        == "Current configuration will not reload as not all conditions are met,"
+        "please refer to documentation."
+    )
+
+    config_no_reload = Config(
+        app="tests.test_config:asgi_app", reload_dirs=[str(tmp_path)]
+    )
+    assert not config_no_reload.should_reload
+    assert len(caplog.records) == 2
+    assert (
+        caplog.records[-1].message
+        == "Current configuration will not reload as not all conditions are met,"
+        "please refer to documentation."
+    )
+
+
+def test_reload_dir_is_set(
+    reload_directory_structure: Path, caplog: pytest.LogCaptureFixture
+) -> None:
+    app_dir = reload_directory_structure / "app"
+    with caplog.at_level(logging.INFO):
+        config = Config(
+            app="tests.test_config:asgi_app", reload=True, reload_dirs=[str(app_dir)]
+        )
+        assert len(caplog.records) == 1
+        assert (
+            caplog.records[-1].message
+            == f"Will watch for changes in these directories: ['{app_dir}']"
+        )
+        assert config.reload_dirs == [app_dir]
+
+
+def test_non_existant_reload_dir_is_not_set(
+    reload_directory_structure: Path, caplog: pytest.LogCaptureFixture
+) -> None:
+    with as_cwd(reload_directory_structure), caplog.at_level(logging.WARNING):
+        config = Config(
+            app="tests.test_config:asgi_app", reload=True, reload_dirs=["reload"]
+        )
+        assert config.reload_dirs == [reload_directory_structure]
+        assert (
+            caplog.records[-1].message
+            == "Provided reload directories ['reload'] did not contain valid "
+            + "directories, watching current working directory."
+        )
+
+
+def test_reload_subdir_removal(reload_directory_structure: Path) -> None:
```

Comment:
```suggestion
def test_duplicate_direcories_are_removed(reload_directory_structure: Path) -> None:
```
not sure about this one I let you choose